### PR TITLE
#519 Show product overview in the popup dialog

### PR DESCRIPTION
**Pull request by justindongqiang**
Please have a look. It is pretty much done. A couple of notes: that I did some refactory to product details so that I could share the method to get product detail. Also, seems  partial view "section head" including  CSS or JS does not appear to be working. I added the script files to layout for now, not ideal but will need find another way to handle this. There is one similar issue for your reference "https://stackoverflow.com/questions/5110028/add-css-or-javascript-files-to-layout-head-from-views-or-partial-views".

### CODE REVIEW

**Comment by justindongqiang** [comment]
The quick view link on product thumb may not look perfect and we may need to decorate it a bit to make it looks prettier.

**Comment by thiennn** [comment]
Sorry I still have got time to look into this. I am trying, hopefully will be soon

File: src/Modules/SimplCommerce.Module.Catalog/Views/Product/ProductOverview.cshtml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+                                    <a href="#" class="add-to-comparison">Add to Compare</a>
+                                </li>
+                            </ul>
+                        </div>
+                    </form>
+                </div>
+            }
+        </div>
+    </div>
+    @if (!string.IsNullOrWhiteSpace(Model.Description))
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Catalog/Views/Product/ProductOverview.cshtml]: I think the product overview should not contain Description, and Specification, these information can be very long

File: src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -105,7 +116,10 @@
     <script simpl-append-version="true" src="~/modules/shopping-cart/add-to-cart.js"></script>
     <script simpl-append-version="true" src="~/modules/product-comparison/product-comparison.js"></script>
     <script simpl-append-version="true" src="~/modules/wish-list/wishlist.js"></script>
-    @RenderSection("scripts", required: false)
+    <script simpl-append-version="true" src="~/modules/catalog/product-overview.js"></script>
+
+
+@RenderSection("scripts", required: false)
```
Comment (thiennn) [inline src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml]: remove empty line and fix the indent

File: src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -14,6 +14,11 @@
     <link simpl-append-version="true" rel="stylesheet" href="~/lib/multi-dropdown-menu/css/bootstrap-4-navbar.css" />
     <link simpl-append-version="true" rel="stylesheet" href="~/modules/core/site.css" />
 
+    @{
+        <link simpl-append-version="true" rel="stylesheet" href="~/lib/smoothproducts/css/smoothproducts.css" />
+        <link simpl-append-version="true" rel="stylesheet" href="~/modules/catalog/product-detail.css" />
+    }
+
```
Comment (thiennn) [inline src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml]: why need to put in this @{ } those file will need in other Layout also ?

File: src/Modules/SimplCommerce.Module.Catalog/Views/Product/ProductOverview.cshtml [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    <meta name="title" content="@ViewBag.Title" />
+    <meta name="keywords" content="@Model.MetaKeywords" />
+    <meta name="description" content="@Model.MetaDescription" />
+}*@
+
+@*@section head
+{
+    <link simpl-append-version="true" rel="stylesheet" href="~/lib/smoothproducts/css/smoothproducts.css" />
+    <link simpl-append-version="true" rel="stylesheet" href="~/modules/catalog/product-detail.css" />
+}*@
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Catalog/Views/Product/ProductOverview.cshtml]: We cannot do this in a partial view. We don't need the meta tag for this circumstance. The css file can be included in the _Layout

File: src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml (hunk 3 above)
Comment (justindongqiang) [reply src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml]: I added the script files to layout for now, just for now,  will need find another way to handle this. 
Partial view is not able to refer to any css files for some reason.There is one similar issue for your reference "https://stackoverflow.com/questions/5110028/add-css-or-javascript-files-to-layout-head-from-views-or-partial-views".

File: src/Modules/SimplCommerce.Module.Catalog/Views/Product/ProductOverview.cshtml (hunk 4 above)
Comment (justindongqiang) [reply src/Modules/SimplCommerce.Module.Catalog/Views/Product/ProductOverview.cshtml]: yes  I could remove it from ProductOverview.cshtml

File: src/Modules/SimplCommerce.Module.Catalog/Views/Product/ProductOverview.cshtml (hunk 1 above)
Comment (justindongqiang) [reply src/Modules/SimplCommerce.Module.Catalog/Views/Product/ProductOverview.cshtml]: sure I removed them

File: src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml (hunk 2 above)
Comment (justindongqiang) [reply src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml]: done

File: src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml (hunk 3 above)
Comment (justindongqiang) [reply src/SimplCommerce.WebHost/Views/Shared/_Layout.cshtml]: done

**Comment by justindongqiang** [comment]
@thiennn all comments resolved and please review.

**Comment by justindongqiang** [comment]
Thanks👌👍

**Pull request merged**